<!DOCTYPE HTML>
<html lang="zh-CN">


<head>
    <meta charset="utf-8">
    <meta name="keywords" content="【生信资源与实践】第二次上机报告, Python Bioinformatics 宅男 书呆子 自言自语">
    <meta name="description" content="一条没有故事的巛">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="renderer" content="webkit|ie-stand|ie-comp">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <!-- Global site tag (gtag.js) - Google Analytics -->


    <title>【生信资源与实践】第二次上机报告 | 风渡巛</title>
    <link rel="icon" type="image/png" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/favicon.png">

    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/awesome/css/all.css">
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/materialize/materialize.min.css">
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/aos/aos.css">
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/animate/animate.min.css">
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/lightGallery/css/lightgallery.min.css">
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/css/matery.css">
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/css/my.css">

    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/jquery/jquery.min.js"></script>

<meta name="generator" content="Hexo 5.2.0"><link rel="alternate" href="/atom.xml" title="风渡巛" type="application/atom+xml">

<style>.github-emoji { position: relative; display: inline-block; width: 1.2em; min-height: 1.2em; overflow: hidden; vertical-align: top; color: transparent; }  .github-emoji > span { position: relative; z-index: 10; }  .github-emoji img, .github-emoji .fancybox { margin: 0 !important; padding: 0 !important; border: none !important; outline: none !important; text-decoration: none !important; user-select: none !important; cursor: auto !important; }  .github-emoji img { height: 1.2em !important; width: 1.2em !important; position: absolute !important; left: 50% !important; top: 50% !important; transform: translate(-50%, -50%) !important; user-select: none !important; cursor: auto !important; } .github-emoji-fallback { color: inherit; } .github-emoji-fallback img { opacity: 0 !important; }</style>
<link rel="stylesheet" href="/css/prism-tomorrow.css" type="text/css"></head>




<body>
    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/artitalk/artitalk.min.css">

<header class="navbar-fixed">
    <nav id="headNav" class="bg-color nav-transparent">
        <div id="navContainer" class="nav-wrapper container">
            <div class="brand-logo">
                <a href="/" class="waves-effect waves-light">
                    <div>
                        
                        <img src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/medias/logo.png" class="logo-img" alt="LOGO">
                        
                        <span class="logo-span">风渡巛</span>
                    </div>

                </a>
            </div>
            

<a href="#" data-target="mobile-nav" class="sidenav-trigger button-collapse"><i class="fas fa-bars"></i></a>
<ul class="right nav-menu">
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/" class="waves-effect waves-light">
      
      <i class="fas fa-home" style="zoom: 0.6;"></i>
      
      <span>首页</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/tags" class="waves-effect waves-light">
      
      <i class="fas fa-tags" style="zoom: 0.6;"></i>
      
      <span>标签</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/categories" class="waves-effect waves-light">
      
      <i class="fas fa-bookmark" style="zoom: 0.6;"></i>
      
      <span>分类</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/archives" class="waves-effect waves-light">
      
      <i class="fas fa-archive" style="zoom: 0.6;"></i>
      
      <span>归档</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/about" class="waves-effect waves-light">
      
      <i class="fas fa-user-circle" style="zoom: 0.6;"></i>
      
      <span>关于</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/contact" class="waves-effect waves-light">
      
      <i class="fas fa-comments" style="zoom: 0.6;"></i>
      
      <span>留言板</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/artitalk" class="waves-effect waves-light">
      
      <i class="fas fa-envelope" style="zoom: 0.6;"></i>
      
      <span>碎语</span>
    </a>
    
  </li>
  
  <li>
    <a href="#searchModal" class="modal-trigger waves-effect waves-light">
      <i id="searchIcon" class="fas fa-search" title="搜索" style="zoom: 0.85;"></i>
    </a>
  </li>
</ul>


<div id="mobile-nav" class="side-nav sidenav">

    <div class="mobile-head bg-color">
        
        <img src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/medias/logo.png" class="logo-img circle responsive-img">
        
        <div class="logo-name">风渡巛</div>
        <div class="logo-desc">
            
            一条没有故事的巛
            
        </div>
    </div>

    

    <ul class="menu-list mobile-menu-list">
        
        <li class="m-nav-item">
	  
		<a href="/" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-home"></i>
			
			首页
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/tags" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-tags"></i>
			
			标签
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/categories" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-bookmark"></i>
			
			分类
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/archives" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-archive"></i>
			
			归档
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/about" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-user-circle"></i>
			
			关于
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/contact" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-comments"></i>
			
			留言板
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/artitalk" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-envelope"></i>
			
			Artitalk
		</a>
          
        </li>
        
        
        <li><div class="divider"></div></li>
        <li>
            <a href="https://github.com/Achuan-2/Achuan-2.github.io" class="waves-effect waves-light" target="_blank">
                <i class="fab fa-github-square fa-fw"></i>Fork Me
            </a>
        </li>
        
    </ul>
</div>


        </div>

        
            <style>
    .nav-transparent .github-corner {
        display: none !important;
    }

    .github-corner {
        position: absolute;
        z-index: 10;
        top: 0;
        right: 0;
        border: 0;
        transform: scale(1.1);
    }

    .github-corner svg {
        color: #0f9d58;
        fill: #fff;
        height: 64px;
        width: 64px;
    }

    .github-corner:hover .octo-arm {
        animation: a 0.56s ease-in-out;
    }

    .github-corner .octo-arm {
        animation: none;
    }

    @keyframes a {
        0%,
        to {
            transform: rotate(0);
        }
        20%,
        60% {
            transform: rotate(-25deg);
        }
        40%,
        80% {
            transform: rotate(10deg);
        }
    }
</style>

<a href="https://github.com/Achuan-2/Achuan-2.github.io" class="github-corner tooltipped hide-on-med-and-down" target="_blank"
   data-tooltip="Fork Me" data-position="left" data-delay="50">
    <svg viewBox="0 0 250 250" aria-hidden="true">
        <path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path>
        <path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
              fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path>
        <path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
              fill="currentColor" class="octo-body"></path>
    </svg>
</a>
        
    </nav>

</header>

    



<div class="bg-cover pd-header post-cover" style="background-image: url('https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/medias/featureimages/4.jpg')">
    <div class="container" style="right: 0px;left: 0px;">
        <div class="row">
            <div class="col s12 m12 l12">
                <div class="brand">
                    <h1 class="description center-align post-title">【生信资源与实践】第二次上机报告</h1>
                </div>
            </div>
        </div>
    </div>
</div>




<main class="post-container content">

    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/tocbot/tocbot.css">
<style>
    #articleContent h1::before,
    #articleContent h2::before,
    #articleContent h3::before,
    #articleContent h4::before,
    #articleContent h5::before,
    #articleContent h6::before {
        display: block;
        content: " ";
        height: 100px;
        margin-top: -100px;
        visibility: hidden;
    }

    #articleContent :focus {
        outline: none;
    }

    .toc-fixed {
        position: fixed;
        top: 64px;
    }

    .toc-widget {
        width: 345px;
        padding-left: 20px;
    }

    .toc-widget .toc-title {
        margin: 35px 0 15px 0;
        padding-left: 17px;
        font-size: 1.5rem;
        font-weight: bold;
        line-height: 1.5rem;
    }

    .toc-widget ol {
        padding: 0;
        list-style: none;
    }

    #toc-content {
        height: calc(100vh - 250px);
        overflow: auto;
    }

    #toc-content ol {
        padding-left: 10px;
    }

    #toc-content ol li {
        padding-left: 10px;
    }

    #toc-content .toc-link:hover {
        color: #42b983;
        font-weight: 700;
        text-decoration: underline;
    }

    #toc-content .toc-link::before {
        background-color: transparent;
        max-height: 25px;

        position: absolute;
        right: 23.5vw;
        display: block;
    }

    #toc-content .is-active-link {
        color: #42b983;
    }

    #floating-toc-btn {
        position: fixed;
        right: 15px;
        bottom: 76px;
        padding-top: 15px;
        margin-bottom: 0;
        z-index: 998;
    }

    #floating-toc-btn .btn-floating {
        width: 48px;
        height: 48px;
    }

    #floating-toc-btn .btn-floating i {
        line-height: 48px;
        font-size: 1.4rem;
    }
</style>
<div class="row">
    <div id="main-content" class="col s12 m12 l9">
        <!-- 文章内容详情 -->
<div id="artDetail">
    <div class="card">
        <div class="card-content article-info">
            <div class="row tag-cate">
                <div class="col s7">
                    
                    <div class="article-tag">
                        
                            <a href="/tags/%E7%94%9F%E4%BF%A1%E8%B5%84%E6%BA%90%E4%B8%8E%E5%AE%9E%E8%B7%B5/">
                                <span class="chip bg-color">生信资源与实践</span>
                            </a>
                        
                    </div>
                    
                </div>
                <div class="col s5 right-align">
                    
                    <div class="post-cate">
                        <i class="fas fa-bookmark fa-fw icon-category"></i>
                        
                            <a href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/" class="post-category">
                                生物信息
                            </a>
                        
                    </div>
                    
                </div>
            </div>

            <div class="post-info">
                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-minus fa-fw"></i>发布日期:&nbsp;&nbsp;
                    2020-12-25
                </div>
                

                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-check fa-fw"></i>更新日期:&nbsp;&nbsp;
                    2020-12-25
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-file-word fa-fw"></i>文章字数:&nbsp;&nbsp;
                    6k
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-clock fa-fw"></i>阅读时长:&nbsp;&nbsp;
                    24 分
                </div>
                

                
                    <div id="busuanzi_container_page_pv" class="info-break-policy">
                        <i class="far fa-eye fa-fw"></i>阅读次数:&nbsp;&nbsp;
                        <span id="busuanzi_value_page_pv"></span>
                    </div>
				
            </div>
        </div>
        <hr class="clearfix">
        <div class="card-content article-card-content">
            <div id="articleContent">
                <h2 id="一、RNA-seq-HISAT-StringTie-ballgown"><a href="#一、RNA-seq-HISAT-StringTie-ballgown" class="headerlink" title="一、RNA-seq (HISAT - StringTie - ballgown)"></a>一、RNA-seq (HISAT - StringTie - ballgown)</h2><p>使用的软件介绍</p>
<ol>
<li><p><strong>HISAT介绍</strong> <a target="_blank" rel="noopener" href="http://ccb.jhu.edu/software/hisat/index.shtml">HISAT官网</a></p>
<p>HISAT全称为Hierarchical Indexing for Spliced Alignment of Transcripts，由约翰霍普金斯大学Steven Salzberg团队开发。它取代Bowtie/TopHat程序，能够将RNA-Seq的读取与基因组进行快速比对，该软件应用了两类不同的索引类型：代表全基因组的<strong>全局FM索引</strong>和大量的<strong>局部小索引</strong>，每个索引代表64000bp。以人类基因组为例，创建了48000个局部索引，每一个覆盖1024bp，最终可以覆盖这个3 billion 的碱基的基因组。这种存在交叉（overlap）的边界可以轻松的比对那些跨区域的read（可变剪切体）。尽管有很多索引，但是hisat会把他们使用合适方法压缩，最终只占4gb左右的内存。本次实验使用的是hisat2版本。</p>
</li>
<li><p><strong>SAMtools</strong></p>
<p>SAMtools是一个用于操作sam和bam文件的工具合集。能够实现二进制查看、格式转换、排序及合并等功能，结合sam格式中的flag、tag等信息，还可以完成比对结果的统计汇总。同时利用linux中的grep、awk等操作命令，还可以大大扩展samtools的使用范围与功能。</p>
</li>
<li><p><strong>StringTie</strong> <a target="_blank" rel="noopener" href="http://www.ccb.jhu.edu/software/stringtie/">StringTie官网</a></p>
<p>基于参考基因组的转录组组装和定量，能够应用流神经网络算法和可选的de novo组装进行转录本组装并预计表达水平。与Cufflinks等程序相比，StringTie实现了更完整、更准确的基因重建，并更好地预测了表达水平。</p>
</li>
<li><p><strong>Ballgown</strong></p>
<p>Ballgown是组装转录组的统计分析工具的R包，用于差异表达分析，转录本结构的可视化以及组装转录本与注释的匹配。</p>
</li>
</ol>
<h3 id="1-1-准备工作，运行RNA-seq-pipeline-sh"><a href="#1-1-准备工作，运行RNA-seq-pipeline-sh" class="headerlink" title="1.1 准备工作，运行RNA_seq_pipeline.sh"></a>1.1 准备工作，运行RNA_seq_pipeline.sh</h3><p><code>mkdir 第二次上机</code>创建文件夹，<code> cp /project/liubl/RNA_seq_pipeline/RNA_seq_prepare.sh  ./</code>将RNA_seq_pipeline.sh 移动到此文件夹中，然后 <code>sh RNA_seq_prepare.sh</code>运行脚本，以创建软件、数据的软链接，拷贝work.sh脚本到当前文件夹</p>
<p><img src="https://cdn.jsdelivr.net/gh/Achuan-2/PicBed/assets/UM1wz93nkLqJxEH.png"></p>
<h3 id="1-2-命名变量，方便之后的命令输入，同时创建文件夹"><a href="#1-2-命名变量，方便之后的命令输入，同时创建文件夹" class="headerlink" title="1.2 命名变量，方便之后的命令输入，同时创建文件夹"></a>1.2 命名变量，方便之后的命令输入，同时创建文件夹</h3><p>在第二次上机文件夹中使用 <code>cd RNA_seq_pipeline/</code>进入RNA_seq_pipeline文件夹</p>
<pre class=" language-bash"><code class="language-bash"><span class="token shebang important">#!/bin/sh</span> -ex
<span class="token comment" spellcheck="true"># 是指此脚本使用/bin/sh 来执行</span></code></pre>
<pre class=" language-bash"><code class="language-bash">-x 调试用,打印每条命令执行情况

-e 一个命令在执行后返回一个非0状态值时即error时,就退出</code></pre>
<p><strong>对变量命名，并创建文件夹</strong></p>
<pre class=" language-bash"><code class="language-bash"><span class="token comment" spellcheck="true">#设置当前文件夹</span>
PROJECT_HOME<span class="token operator">=</span><span class="token variable">$PWD</span>
<span class="token comment" spellcheck="true">#设置INDEX，DATA，RESULT</span>
INDEX<span class="token operator">=</span><span class="token variable">$PROJECT_HOME</span>/index
DATA<span class="token operator">=</span><span class="token variable">$PROJECT_HOME</span>/data
RESULT<span class="token operator">=</span><span class="token variable">$PROJECT_HOME</span>/results
<span class="token comment" spellcheck="true">#创建文件夹</span>
<span class="token function">mkdir</span> -p <span class="token variable">$INDEX</span>
<span class="token function">mkdir</span> -p <span class="token variable">$RESULT</span>
<span class="token function">mkdir</span> -p <span class="token variable">$RESULT</span>/ballgown</code></pre>
<p><img src="https://i.loli.net/2020/12/25/Zhku67yNF9fRarg.png"></p>
<h3 id="1-3-创建索引"><a href="#1-3-创建索引" class="headerlink" title="1.3 创建索引"></a>1.3 创建索引</h3><pre class=" language-bash"><code class="language-bash"><span class="token comment" spellcheck="true"># build hisat2 index</span>
hisat/hisat2-build <span class="token variable">$DATA</span>/chrX.fa <span class="token variable">$INDEX</span>/chr_index
<span class="token function">wait</span></code></pre>
<p><img src="https://i.loli.net/2020/12/25/ZrhOdERgKuXkBw9.png"></p>
<p>运行结束</p>
<p><img src="https://i.loli.net/2020/12/25/dqn84YC25JUPsMj.png"></p>
<blockquote>
<p>为什么要用index？</p>
<p>hisat的index使用的是FM索引,FM 索引基于BWT算法，FM-index包括三部分:</p>
<p>①BWT（T），对于第一列只需要用数字分别表示ACGT的个数就好;last column 用00，01，10，11来记录ATCG; </p>
<p>②checkpoint data，每隔128位,提前计算每个ATCG的重复次数</p>
<p>③SA sample ，每32个位置设定读段所在的位置，仅需要回读到sample位置即可知道读段位置</p>
<p>相比其他构建索引方式（后缀树等）占用内存更小，有利于短读段快速回帖参考基因组上.</p>
<p><img src="https://i.loli.net/2020/12/25/Nf1zHSs9mL4ZFJd.png"></p>
</blockquote>
<h3 id="1-4-用hisat2将reads比对到基因组"><a href="#1-4-用hisat2将reads比对到基因组" class="headerlink" title="1.4 用hisat2将reads比对到基因组"></a>1.4 用hisat2将reads比对到基因组</h3><p>输入命令</p>
<pre class=" language-bash"><code class="language-bash">hisat/hisat2 -x <span class="token variable">$INDEX</span>/chr_index -1  <span class="token variable">$DATA</span>/ERR188044_chrX_1.fastq -2 <span class="token variable">$DATA</span>/ERR188044_chrX_2.fastq -S <span class="token variable">$RESULT</span>/ERR188044_chrX_reads.sam <span class="token operator">&amp;</span>
hisat/hisat2 -x <span class="token variable">$INDEX</span>/chr_index -1  <span class="token variable">$DATA</span>/ERR188104_chrX_1.fastq -2 <span class="token variable">$DATA</span>/ERR188104_chrX_2.fastq -S <span class="token variable">$RESULT</span>/ERR188104_chrX_reads.sam <span class="token operator">&amp;</span>
hisat/hisat2 -x <span class="token variable">$INDEX</span>/chr_index -1  <span class="token variable">$DATA</span>/ERR188234_chrX_1.fastq -2 <span class="token variable">$DATA</span>/ERR188234_chrX_2.fastq -S <span class="token variable">$RESULT</span>/ERR188234_chrX_reads.sam <span class="token operator">&amp;</span>
hisat/hisat2 -x <span class="token variable">$INDEX</span>/chr_index -1  <span class="token variable">$DATA</span>/ERR188245_chrX_1.fastq -2 <span class="token variable">$DATA</span>/ERR188245_chrX_2.fastq -S <span class="token variable">$RESULT</span>/ERR188245_chrX_reads.sam <span class="token operator">&amp;</span>
hisat/hisat2 -x <span class="token variable">$INDEX</span>/chr_index -1  <span class="token variable">$DATA</span>/ERR188257_chrX_1.fastq -2 <span class="token variable">$DATA</span>/ERR188257_chrX_2.fastq -S <span class="token variable">$RESULT</span>/ERR188257_chrX_reads.sam <span class="token operator">&amp;</span>
hisat/hisat2 -x <span class="token variable">$INDEX</span>/chr_index -1  <span class="token variable">$DATA</span>/ERR188273_chrX_1.fastq -2 <span class="token variable">$DATA</span>/ERR188273_chrX_2.fastq -S <span class="token variable">$RESULT</span>/ERR188273_chrX_reads.sam <span class="token operator">&amp;</span>
<span class="token function">wait</span></code></pre>
<pre class=" language-bash"><code class="language-bash">-x 是指我们之前构建的参考基因组的位置和前缀
-1 是指样本的R1文件
-2 是指样本的R2文件
-S 是指输出文件的名字和格式，一般使用sam格式
<span class="token operator">&amp;</span> 后面加<span class="token operator">&amp;</span>是为了能够并行</code></pre>
<p>比对结果，共六个，以第二个为例进行分析：</p>
<p>第一部分描述的是pair-end模式（双端测序）下的一致比对结果：</p>
<ul>
<li><p>aligned concordantly就是read1和read2同时合理的比对到了基因组/转录组上。concordantly 0 times见第二部分</p>
</li>
<li><p>aligned concordantly exactly 1 time，exactly 1 time 就是只有一种比对结果。</p>
</li>
<li><p>&gt;1 times 就是read1和read2可以同时比对到多个地方。</p>
</li>
</ul>
<p>第二部分，pair-end模式下不一致的比对结果。</p>
<ul>
<li>concordantly 0 times就是read1和read2不能同时合理的同时比对到基因组/转录组上</li>
<li>aligned discordantly 1 time，discordantly比对就是read1和read2都能比对上，但是不合理（比对方向不对等）</li>
</ul>
<p>第三部分就是对剩余reads（既不能concordantly，也不是discordantly 1 time）的单端模式的比对</p>
<p><img src="https://i.loli.net/2020/12/25/rqEKHdpl9Cc7jOM.png"></p>
<h3 id="1-5-使用samtools进行排序和格式转换"><a href="#1-5-使用samtools进行排序和格式转换" class="headerlink" title="1.5 使用samtools进行排序和格式转换"></a>1.5 使用samtools进行排序和格式转换</h3><pre class=" language-bash"><code class="language-bash">samtools <span class="token function">sort</span> -@ 8 -o <span class="token variable">$RESULT</span>/ERR188044_chrX_reads.bam <span class="token variable">$RESULT</span>/ERR188044_chrX_reads.sam <span class="token operator">&amp;</span>
samtools <span class="token function">sort</span> -@ 8 -o <span class="token variable">$RESULT</span>/ERR188104_chrX_reads.bam <span class="token variable">$RESULT</span>/ERR188104_chrX_reads.sam <span class="token operator">&amp;</span>
samtools <span class="token function">sort</span> -@ 8 -o <span class="token variable">$RESULT</span>/ERR188234_chrX_reads.bam <span class="token variable">$RESULT</span>/ERR188234_chrX_reads.sam <span class="token operator">&amp;</span>
samtools <span class="token function">sort</span> -@ 8 -o <span class="token variable">$RESULT</span>/ERR188245_chrX_reads.bam <span class="token variable">$RESULT</span>/ERR188245_chrX_reads.sam <span class="token operator">&amp;</span>
samtools <span class="token function">sort</span> -@ 8 -o <span class="token variable">$RESULT</span>/ERR188257_chrX_reads.bam <span class="token variable">$RESULT</span>/ERR188257_chrX_reads.sam <span class="token operator">&amp;</span>
samtools <span class="token function">sort</span> -@ 8 -o <span class="token variable">$RESULT</span>/ERR188273_chrX_reads.bam <span class="token variable">$RESULT</span>/ERR188273_chrX_reads.sam <span class="token operator">&amp;</span></code></pre>
<pre class=" language-bash"><code class="language-bash">sort命令格式<span class="token variable"><span class="token variable">`</span> samtools <span class="token function">sort</span> <span class="token punctuation">[</span>-l level<span class="token punctuation">]</span> <span class="token punctuation">[</span>-m maxMem<span class="token punctuation">]</span> <span class="token punctuation">[</span>-o out.bam<span class="token punctuation">]</span> <span class="token punctuation">[</span>-O format<span class="token punctuation">]</span> <span class="token punctuation">[</span>-n<span class="token punctuation">]</span> <span class="token punctuation">[</span>-T tmpprefix<span class="token punctuation">]</span> <span class="token punctuation">[</span>-@ threads<span class="token punctuation">]</span> <span class="token punctuation">[</span>in.sam<span class="token operator">|</span>in.bam<span class="token operator">|</span>in.cram<span class="token punctuation">]</span><span class="token variable">`</span></span>

sort进行排序

-@ 8 设置排序和压缩是的线程数量，默认是单线程。这里设置为8线程

-o 设置输出文件和位置

<span class="token variable">$RESULT</span>/ERR188273_chrX_reads.sam  输入sam文件

<span class="token operator">&amp;</span> 允许并行
</code></pre>
<p>输入命令，将sam文件进行排序，并输出为bam文件</p>
<p><img src="https://i.loli.net/2020/12/25/9UwVBcGqWDCQPez.png"></p>
<h3 id="1-6-使用stringtie对每个样本进行转录本组装"><a href="#1-6-使用stringtie对每个样本进行转录本组装" class="headerlink" title="1.6 使用stringtie对每个样本进行转录本组装"></a>1.6 使用stringtie对每个样本进行转录本组装</h3><p>比对上的reads将会被呈递给StringTie进行<strong>转录本组装</strong>，会针对每个bam文件生成一个gtf文件，它主要记录了转录本的组装信息</p>
<p>输入命令</p>
<pre class=" language-bash"><code class="language-bash">stringtie/stringtie -p 8 -G <span class="token variable">$DATA</span>/chrX.gtf -o <span class="token variable">$RESULT</span>/ERR188044.gtf <span class="token variable">$RESULT</span>/ERR188044_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -p 8 -G <span class="token variable">$DATA</span>/chrX.gtf -o <span class="token variable">$RESULT</span>/ERR188104.gtf <span class="token variable">$RESULT</span>/ERR188104_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -p 8 -G <span class="token variable">$DATA</span>/chrX.gtf -o <span class="token variable">$RESULT</span>/ERR188234.gtf <span class="token variable">$RESULT</span>/ERR188234_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -p 8 -G <span class="token variable">$DATA</span>/chrX.gtf -o <span class="token variable">$RESULT</span>/ERR188245.gtf <span class="token variable">$RESULT</span>/ERR188245_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -p 8 -G <span class="token variable">$DATA</span>/chrX.gtf -o <span class="token variable">$RESULT</span>/ERR188257.gtf <span class="token variable">$RESULT</span>/ERR188257_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -p 8 -G <span class="token variable">$DATA</span>/chrX.gtf -o <span class="token variable">$RESULT</span>/ERR188273.gtf <span class="token variable">$RESULT</span>/ERR188273_chrX_reads.bam <span class="token operator">&amp;</span></code></pre>
<pre class=" language-bash"><code class="language-bash">-p  8    指定组装转录本的线程数（CPU）。默认值是1, 这里指定为1

-G 参数指定基因组注释文件，

-o 输出的 gtf 路径

<span class="token variable">$RESULT</span>/ERR188273_chrX_reads.bam <span class="token operator">&amp;</span> <span class="token keyword">:</span> 输入bam文件</code></pre>
<p><img src="https://i.loli.net/2020/12/25/1JEpGKP6uIeozgF.png"></p>
<p>使用<code>head -10 ERR188234.gtf</code>查看GTF文件<br><img src="https://i.loli.net/2020/12/25/LzqMkfIPYZNu9vU.png"></p>
<blockquote>
<p><strong>GTF文件：记录组装的转录本信息</strong></p>
<ul>
<li><p><strong>seqname</strong>: 染色体，contig, 或 scaffold</p>
</li>
<li><p><strong>source</strong>: GTF文件的源文件。</p>
</li>
<li><p><strong>feature</strong>: 特征类型；如：exon, transcript, mRNA, 5’UTR。</p>
</li>
<li><p><strong>start</strong>: 开始位置，使用基于1的索引</p>
</li>
<li><p><strong>end</strong>: 结束位置，使用基于1的索引</p>
</li>
<li><p><strong>score</strong>: 组装的转录本的可信度分数。如果转录本 与a read alignment bundle有连接，则StringTie输出常数值1000。</p>
</li>
<li><p><strong>strand</strong>: 正向链： ‘+’； 反向链： ‘-‘.</p>
</li>
<li><p><strong>frame</strong>: CDS特征的 Frame or phase 。 StringTie不使用该字段，只记录一个“.”。</p>
</li>
<li><p><strong>attributes</strong>:</p>
</li>
<li><p><strong>gene_id</strong>: A unique identifier for a single gene and its child transcript and exons based on the alignments’ file name.</p>
</li>
<li><p><strong>transcript_id</strong>: A unique identifier for a single transcript and its child exons based on the alignments’ file name.</p>
</li>
<li><p><strong>exon_number</strong>: A unique identifier for a single exon, starting from 1, within a given transcript.</p>
</li>
<li><p><strong>reference_id</strong>: The transcript_id in the reference annotation (optional) that the instance matched.</p>
</li>
<li><p><strong>ref_gene_id</strong>: The gene_id in the reference annotation (optional) that the instance matched.</p>
</li>
<li><p><strong>ref_gene_name</strong>: The gene_name in the reference annotation (optional) that the instance matched.</p>
</li>
<li><p><strong>cov</strong>: The average per-base coverage for the transcript or exon.</p>
</li>
<li><p><strong>FPKM</strong>: <strong>Fragments per kilobase of transcript per million read pairs</strong>. This is the number of pairs of reads aligning to this feature, normalized by the total number of fragments sequenced (in millions) and the length of the transcript (in kilobases).</p>
</li>
<li><p><strong>TPM</strong>: <strong>Transcripts per million</strong>. This is the number of transcripts from this particular gene normalized first by gene length, and then by sequencing depth (in millions) in the sample. A detailed explanation and a comparison of TPM and FPKM can be found <a target="_blank" rel="noopener" href="http://www.rna-seqblog.com/rpkm-fpkm-and-tpm-clearly-explained/">here</a>, and TPM was defined <a target="_blank" rel="noopener" href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323">by B. Li and C. Dewey here</a>.</p>
</li>
</ul>
</blockquote>
<h3 id="1-7-stringtie合并转录本"><a href="#1-7-stringtie合并转录本" class="headerlink" title="1.7 stringtie合并转录本"></a>1.7 stringtie合并转录本</h3><p>因为有些样本的转录本可能仅仅被部分reads覆盖，无法被第二步的StringTie组装出来。merge步骤可以创建出所有样本里面都有的转录本，方便下一步的对比。在合并模式下，stringtie将所有样品的GTF为文件作为输入，并将这些转录本组装成非冗余的转录本集合，用以生成一个跨多个RNA-seq样品的全局的、统一的转录本。</p>
<p>输入命令</p>
<pre class=" language-bash"><code class="language-bash">stringtie/stringtie --merge -p 8 -G  <span class="token variable">$DATA</span>/chrX.gtf -o stringtie_merged.gtf  <span class="token variable">$RESULT</span>/*.gtf·</code></pre>
<pre class=" language-bash"><code class="language-bash">-- merge 合并

-p 线程

-G 注释文件

-o 输出

<span class="token variable">$RESULT</span>/*.gtf  输入上一步每个样本的转录文件</code></pre>
<p><img src="https://i.loli.net/2020/12/25/SPu9onz7YTf6btW.png"></p>
<p>查看结果，得到的<code>stringtie_merged.gtf</code>就在RNA_seq_pipeline目录下，使用 <code>head -10 stringtie_merged.gtf</code>预览</p>
<p><img src="https://i.loli.net/2020/12/25/qJIbe7urYjUcZ2z.png"></p>
<h3 id="1-8-使用stringle计算表达量并且为Ballgown包提供输入文件"><a href="#1-8-使用stringle计算表达量并且为Ballgown包提供输入文件" class="headerlink" title="1.8 使用stringle计算表达量并且为Ballgown包提供输入文件"></a>1.8 使用stringle计算表达量并且为Ballgown包提供输入文件</h3><pre class=" language-bash"><code class="language-bash">stringtie/stringtie -e -B -p 8 -G stringtie_merged.gtf -o <span class="token variable">$RESULT</span>/ballgown/ERR188044/ERR188044.gtf <span class="token variable">$RESULT</span>/ERR188044_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -e -B -p 8 -G stringtie_merged.gtf -o <span class="token variable">$RESULT</span>/ballgown/ERR188104/ERR188104.gtf <span class="token variable">$RESULT</span>/ERR188104_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -e -B -p 8 -G stringtie_merged.gtf -o <span class="token variable">$RESULT</span>/ballgown/ERR188234/ERR188234.gtf <span class="token variable">$RESULT</span>/ERR188234_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -e -B -p 8 -G stringtie_merged.gtf -o <span class="token variable">$RESULT</span>/ballgown/ERR188245/ERR188245.gtf <span class="token variable">$RESULT</span>/ERR188245_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -e -B -p 8 -G stringtie_merged.gtf -o <span class="token variable">$RESULT</span>/ballgown/ERR188257/ERR188257.gtf <span class="token variable">$RESULT</span>/ERR188257_chrX_reads.bam <span class="token operator">&amp;</span>
stringtie/stringtie -e -B -p 8 -G stringtie_merged.gtf -o <span class="token variable">$RESULT</span>/ballgown/ERR188273/ERR188273.gtf <span class="token variable">$RESULT</span>/ERR188273_chrX_reads.bam <span class="token operator">&amp;</span></code></pre>
<pre class=" language-bash"><code class="language-bash">-e:用于指定是否仅为参考转录本估计表达丰度；

-B:用于指定是否输出 Ballgown table 文件；

-p: 用于指定线程数<span class="token punctuation">;</span>

-G :用于指定已组装的注释文件；

-o:用于指定输出结果的文件名；</code></pre>
<p> 运行命令</p>
<p><img src="https://i.loli.net/2020/12/25/AwktQBZgD25IVbM.png"></p>
<p><img src="https://i.loli.net/2020/12/25/6AuvlJWK3VhRm1z.png"></p>
<p>查看<code>$RESULT/ballgown/ERR188234/ERR188234.gtf</code></p>
<p><img src="https://i.loli.net/2020/12/25/MKmBjClr1hxzAST.png" alt="image-20201023163606232"></p>
<h3 id="1-9-Ballgown表达量分析"><a href="#1-9-Ballgown表达量分析" class="headerlink" title="1.9  Ballgown表达量分析"></a>1.9  Ballgown表达量分析</h3><pre class=" language-bash"><code class="language-bash">Rscript ./RunBallgown.r</code></pre>
<p>输入命令</p>
<p><img src="https://i.loli.net/2020/12/25/kwyxbJVQcm941GT.png"></p>
<p><img src="https://i.loli.net/2020/12/25/z9ReZmsfWXPyunr.png"></p>
<p>使用<code>ls</code>命令，发现多了一个result文件，将文件通过FileZilla软件传输到个人电脑上，查看result.pdf文件</p>
<p><img src="https://i.loli.net/2020/12/25/OUPLF2yBmfYo1Hl.png"></p>
<p><img src="https://i.loli.net/2020/12/25/FA68xmvhs3HZN4r.png"></p>
<p><img src="https://i.loli.net/2020/12/25/WJAQBNwqsbpj64T.png"></p>
<h3 id="1-10-使用IGV查看比对情况"><a href="#1-10-使用IGV查看比对情况" class="headerlink" title="1.10 使用IGV查看比对情况"></a>1.10 使用IGV查看比对情况</h3><p>输入下面命令，使用samtools对bam建立索引，得到bai文件</p>
<pre class=" language-bash"><code class="language-bash"><span class="token comment" spellcheck="true">#samtools 对BAM文件建立索引</span>
samtools index ERR188044_chrX_reads.bam <span class="token operator">&amp;</span>
samtools index ERR188104_chrX_reads.bam <span class="token operator">&amp;</span>
samtools index ERR188234_chrX_reads.bam <span class="token operator">&amp;</span>
samtools index ERR188245_chrX_reads.bam <span class="token operator">&amp;</span>
samtools index ERR188257_chrX_reads.bam <span class="token operator">&amp;</span>
samtools index ERR188273_chrX_reads.bam <span class="token operator">&amp;</span></code></pre>
<p><img src="https://i.loli.net/2020/12/25/sNLTXbWVlHYy28o.png"></p>
<p>导入ERR188004_chr_reads.bam，选择参考基因组hg19 chrX，位置chrX:10,143,411-10,143,552，得到如下图结果</p>
<p><img src="https://i.loli.net/2020/12/25/OUhCcGRP5Zkl7Jq.png"></p>
<blockquote>
<p>绿红蓝棕色分别代表A、T、C、G，灰色代表序列和参考序列一致，若不一致coverage将以ATCG对应颜色显示</p>
</blockquote>
<p>发现大部分读段没有mapping到外显子上，才知道这次参考基因组使用的hg38，于是改为hg38</p>
<p><img src="https://i.loli.net/2020/12/25/cNZ9oOmlIG14QaR.png"></p>
<p>然后导入ERR188004_chr_reads.bam（需要把bai文件放在同一文件夹下），选择参考基因组hg38 chrX，位置chrX:10,135,678-10,144,820，接下来查看其他样本同一位置的比对情况</p>
<p><img src="https://i.loli.net/2020/12/25/vrWlo68ezhikSwa.png"></p>
<p>导入ERR188104_chr_reads.bam </p>
<p><img src="https://i.loli.net/2020/12/25/sGp7BofzDURHyFh.png"></p>
<p>剩下bam文件如上导入，不再赘述，可以看到不同转录本的表达差异。</p>
<h2 id="二、Gene-Assembly"><a href="#二、Gene-Assembly" class="headerlink" title="二、Gene Assembly"></a>二、Gene Assembly</h2><h3 id="基本概念"><a href="#基本概念" class="headerlink" title="基本概念"></a>基本概念</h3><p>由于仪器测序读长的限制等，在建库时会将DNA随机打断为小片段的序列，因此，基因组组装就是将小片段的序列连接起来，但是序列之间的联系十分复杂，常用构建Graph来进行表示，然后在对Graph进行简化、拼接，即<strong>reads→contigs→scaffolds</strong>。</p>
<ol>
<li><strong>ALLPATHS-LG</strong></li>
</ol>
<p><a target="_blank" rel="noopener" href="http://www.broadinstitute.org/software/allpaths-lg/blog/">ALLPATHS-LG</a>是一个基因组组装软件，适合于组装short reads数据，由Computational Research and Development group at the Broad Institute开发。ALLPATHS-LG是现在行业内公认进行基因组<em>De novo</em>（从头组装）组装效果最好的软件,适合于短reads数据。</p>
<p>和其他组装软件不同的是，allpaths-lg要求<strong>至少两个文库</strong>：</p>
<p>第一个文库的插入片段长度不能超过测序读长的两倍，这样可以保证双端测序的reads之间存在overlap，这样的文库类型称之为<strong>fragment</strong></p>
<p>第二个文库的插入片段通常大于3kb,超长读长有利于基因组的组装，这样的文库类型称之为<strong>jumping.</strong></p>
<p>除了插入片段外，allpaths-lg对测序深度也有要求，推荐100X以上。</p>
<p>AllPATHS-LG使用<strong>de Bruijin Graph（DBG）</strong>算法，这是目前常用的二代测序拼接算法。</p>
<p><strong>其步骤：</strong></p>
<p>1）构建DBG图，将reads分割为一系列连续的kmer；所谓kmer，即为一段短的DNA片段。<strong>k为一个奇数</strong>，k等于几，就称为几mer。例如一段7bp的序列，则可以称为7mer。kmer必须为奇数，根本原因是为了避免正反义链混淆。比如 ：5-mer的 CGCGC，反向互补后是 GCGCG， 它们是不同的；这就不会像 4-mer，CGCG发现它反向互补后仍然是CGCG。<strong>k值越大</strong>可辨别更多的小重复序列，越容易把DBG转换为唯一的序列，但得到的拼接过程含有更多的gaps；<strong>小的k值</strong>对应的DBG能够得到较好的连通性，但是算法的复杂度会提高，repeats序列处理会更复杂，增加了错拼的可能性。</p>
<p>2）合并DBG图；</p>
<p>3）构建contig：寻找最优路径（经过每一个节点且仅经过一次，即欧拉行走），最优路径对应的碱基序列构成一个contig；</p>
<p>4)构建scaffold：通过<strong>PE reads</strong>( paired-end <em>reads</em>)位置确定contig之间的相对位置和方向，组装contig，填充contig之间的gap，得到scaffold序列。</p>
<p><img src="https://i.loli.net/2020/12/25/vbZK6t58GTgilNo.png"></p>
<p><img src="https://i.loli.net/2020/12/25/AMBrSnd2vKiLIHh.png"></p>
<ol start="2">
<li><p>GCE </p>
<p>GCE(Genome Characteristics Estimation) 是华大基因用于<a target="_blank" rel="noopener" href="https://www.plob.org/tag/%E5%9F%BA%E5%9B%A0%E7%BB%84">基因组</a>评估的软件，</p>
<p>其文献为：<a target="_blank" rel="noopener" href="http://www.researchgate.net/publication/255722390_Estimation_of_genomic_characteristics_by_analyzing_k-mer_frequency_in_de_novo_genome_projects">Estimation of genomic characteristics by analyzing k-mer frequency in de novo genome projects</a>。</p>
<p>Github地址：<a target="_blank" rel="noopener" href="https://github.com/BioInfoTools/GCE">https://github.com/BioInfoTools/GCE</a></p>
<p>GCE 软件包中主要包含 kmer_freq_hash 和 gce 两支程序。前者用于进行 kmer 的频数统计，后者在前者的结果上进行基因组大小的准确估算。</p>
</li>
<li><p>FASTQC</p>
<p>FastQC是一款基于Java的软件，一般都是在linux环境下使用命令行运行，它可以快速多线程地对测序数据进行质量评估（Quality Control）</p>
<p>拿到原始数据后我们会首先采用fastqc程序进行质控，看原始数据质量情况，fastqc会生成一个html结果报告，根据图形化界面，我们可以判断下机数据情况是否符合分析要求。但fastqc 软件主要是针对全基因组测序的，并且各建库方法不同，其评判标准也会有所差距；<strong>不能只是一味的寻求全部结果通过</strong>。</p>
</li>
</ol>
<h3 id="2-1-环境变量配置"><a href="#2-1-环境变量配置" class="headerlink" title="2.1 环境变量配置"></a>2.1 环境变量配置</h3><pre class=" language-bash"><code class="language-bash"><span class="token comment" spellcheck="true">#before using allpath, you have add /project/hts-demo/genome_assembly/bin/gcc-4.9.1/bin/bin/</span>
<span class="token comment" spellcheck="true"># and /project/hts-demo/genome_assembly/bin/allpathslg/bin/ to you local profile, because allpath relies on one gcc that this ubuntu version can not support.</span>
<span class="token comment" spellcheck="true">#so you have to add gcc to your local profile</span>
<span class="token function">export</span> LD_LIBRARY_PATH<span class="token operator">=</span>/project/hts-demo/genome_assembly/bin/gcc-4.9.1/bin/lib64/:<span class="token variable">$LD_LIBRARY_PATH</span>
PATH<span class="token operator">=</span><span class="token string">"/project/hts-demo/genome_assembly/bin/gcc-4.9.1/bin/bin/:/project/hts-demo/genome_assembly/bin/allpathslg/bin/:<span class="token variable">$PATH</span>"</span></code></pre>
<p>新版的ALLPATHS-LG需求gcc 4.7.0及以上版本，这里将gcc加入环境变量中，使得allpath能够编译程序</p>
<p><img src="https://i.loli.net/2020/12/25/PNvcEzQH27wesDK.png"></p>
<h3 id="2-2-变量命名"><a href="#2-2-变量命名" class="headerlink" title="2.2  变量命名"></a>2.2  变量命名</h3><pre class=" language-bash"><code class="language-bash">PROJECT_HOME<span class="token operator">=</span><span class="token string">"<span class="token variable">$PWD</span>"</span>
BIN<span class="token operator">=</span><span class="token variable">$PROJECT_HOME</span>/bin
DATA<span class="token operator">=</span><span class="token variable">$PROJECT_HOME</span>/data
RESULT<span class="token operator">=</span><span class="token variable">$PROJECT_HOME</span>/result
GCE<span class="token operator">=</span><span class="token variable">$RESULT</span>/gce
FASTQC<span class="token operator">=</span><span class="token variable">$RESULT</span>/fastqc
ALLPATH<span class="token operator">=</span><span class="token variable">$RESULT</span>/allpathlg</code></pre>
<p><img src="https://i.loli.net/2020/12/25/xw3NAC6SZP5iFVs.png"></p>
<p>查看下bin文件夹下有什么东西</p>
<p><img src="https://i.loli.net/2020/12/25/BSEQ2wPAjMTCmVq.png"></p>
<p>绿色高亮的kmer_freq_hash和kmer_freq_pread文件后面有带*，说明是可执行文件</p>
<p>绿色荧光高亮的gce文件，是因为linux系统认为这是一个高风险的目录文件，因为任何人都可以进入到该目录并进行读写入操作</p>
<p><img src="https://i.loli.net/2020/12/25/GBsaeXmFyh78MQN.png"></p>
<h3 id="2-3-GCE估计基因组大小和kmer分布"><a href="#2-3-GCE估计基因组大小和kmer分布" class="headerlink" title="2.3 GCE估计基因组大小和kmer分布"></a>2.3 GCE估计基因组大小和kmer分布</h3><p>①输入命令</p>
<pre class=" language-bash"><code class="language-bash"><span class="token variable">$BIN</span>/kmer_freq_pread -k 17 -l <span class="token variable">$DATA</span>/frags.list -q 33 -m 1  -p <span class="token variable">$GCE</span>/frags.17-kmer 2<span class="token operator">></span> <span class="token variable">$GCE</span>/kmer_freq_pread <span class="token operator">&amp;</span><span class="token operator">>></span>/dev/null</code></pre>
<p>运行<code>kmer_freq_pread</code>查看参数</p>
<p><img src="https://i.loli.net/2020/12/25/U2HSB7PL56qh9tx.png"></p>
<p>运行<code>kmer_freq_hash</code>查看参数, 发现kmer_freq_pread其实是kmer_freq_hash的一个版本</p>
<p><img src="https://i.loli.net/2020/12/25/fowjhObIp1FQm6z.png"></p>
<pre class=" language-bash"><code class="language-bash"><span class="token comment" spellcheck="true">#命令解释</span>
-k 17 设置kmer大小， 默认值为17，kmer的数量和基因序列有数量关系从而评估基因组的大小

-l frags.list    设置我们制作好的输入文件，文本文件，其中每行为一个fastaq文件路径

-q 33     设置质量值的ascii偏移，默认是64，我们的数据是illumina1.8+所以是33

-m  估算模型的选择，离散型0，连续型1，默认为0，对真实数据推荐选择1

-p <span class="token variable">$GCE</span>/frags.17-kmer  设置输出文件的前缀

2<span class="token operator">></span> <span class="token variable">$GCE</span>/kmer_freq_pread <span class="token operator">&amp;</span><span class="token operator">>></span>/dev/null  2<span class="token operator">></span>输出日志信息到文件，，<span class="token operator">&amp;</span>可以代表任意（标准输入输出或错误），<span class="token operator">>></span>代表追加 <span class="token operator">&amp;</span><span class="token operator">>></span>/dev/null就是把标准输出信息和错误信息都重定向给/dev/null。/dev/null是一个特殊的设备文件，这个文件接收到任何数据都会被丢弃。因此，null这个设备通常也被称为位桶（bit bucket）或黑洞。</code></pre>
<p><img src="https://i.loli.net/2020/12/25/4WmQSRe2s19hxuN.png"></p>
<p>②查看结果</p>
<p>使用<code> ls $GCE</code>, 前两个为结果输出文件<img src="https://i.loli.net/2020/12/25/sQijnYNSdAVRveD.png"></p>
<p>使用<code>cat $GCE/kmer_freq_pread</code> 查看$GCE/kmer_freq_pread ，发现为空</p>
<p><img src="https://i.loli.net/2020/12/25/NQiE4So6gdW9vpX.png"></p>
<pre><code>但如果把代码改为`$BIN/kmer_freq_pread -k 17 -l $DATA/frags.list -q 33 -m 1  -p $GCE/frags.17-kmer 2&gt; $GCE/kmer_freq_pread `, 即去掉后面的`&amp;&gt;&gt;/dev/null`, 再使用`cat $GCE/kmer_freq_pread` 查看$GCE/kmer_freq_pread 将得到如下图内容：</code></pre>
<p><img src="https://i.loli.net/2020/12/25/vGkRdIArSEmNKUz.png"></p>
<p><img src="https://i.loli.net/2020/12/25/bxiSpCU2mecWK1g.png"></p>
<p>所以kmer_freq_pread其实是一个log日志文件。若去掉代码中的2&gt; 则直接将运行结果输出</p>
<p>使用 <code> less $GCE/frags.17-kmer.freq.stat</code> 查看frags.17-kmer.freq.stat信息。kmer_freq_hash 的主要结果文件为 species.freq.stat。该文件有 2 列：第1列是kmer重复的次数，第二列是kmer的种类数。该文件有255行，第225行表示kmer重复次数&gt;=255的kmer的总的种类数。</p>
<p><img src="https://i.loli.net/2020/12/25/xvkRZQW3ACzuIro.png"></p>
<p><img src="https://i.loli.net/2020/12/25/hW2yRftOrEwoNSl.png"></p>
<p>使用<code>gzip  -d frags.17-kmer.freq.gz</code>解压frags.17-kmer.freq.gz</p>
<p>使用<code>less $GCE/frags.17-kmer.freq</code>查看解压出来的文件</p>
<p><img src="https://i.loli.net/2020/12/25/SDpn7Gl6mKMUBzr.png"></p>
<p>这是对所有的kmer出现次数的一个统计</p>
<p><img src="https://i.loli.net/2020/12/25/6MwVIypHB3m2vad.png"></p>
<h3 id="2-4-使用FASTQC进行质量控制"><a href="#2-4-使用FASTQC进行质量控制" class="headerlink" title="2.4 使用FASTQC进行质量控制"></a>2.4 使用FASTQC进行质量控制</h3><pre class=" language-bash"><code class="language-bash"><span class="token variable">$BIN</span>/fastqc/fastqc -o <span class="token variable">$FASTQC</span> -f fastq <span class="token variable">$DATA</span>/frags.A.fastq <span class="token variable">$DATA</span>/frags.B.fastq <span class="token operator">&amp;</span> <span class="token operator">>></span>/dev/null</code></pre>
<p>参数介绍</p>
<pre class=" language-bash"><code class="language-bash">-o，结果输出路径；
-f --format 有效的格式是bam、sam、bam_mapped、sam_mapped和fastq，导入两个fastq是因为双端测序
<span class="token operator">&amp;</span> <span class="token operator">>></span>/dev/null 把标准输出和错误都重定向给黑洞</code></pre>
<p>发现这里&amp;和&gt;&gt;间多了一个空格，所以依然会输出过程：</p>
<p><img src="https://i.loli.net/2020/12/25/cXlbmzNVgq2pZIR.png"></p>
<p>结果为result/fastqc/*.html文件，打开fileZilla，有两个html文件将html文件传到个人电脑用浏览器打开(zip压缩包其实就是html各部分内容)</p>
<p><img src="https://i.loli.net/2020/12/25/HFyErS5RQZYiw7c.png"></p>
<p>①frags.A_fastqc.html</p>
<p>结果分为绿色的”PASS”，黄色的”WARN”和红色的”FAIL”。</p>
<p><img src="https://i.loli.net/2020/12/25/ypoiCNhtPuq6aBW.png"></p>
<p>1)基本信息</p>
<p><img src="https://i.loli.net/2020/12/25/LqmWSzK8gEZN35I.png"></p>
<p>2)Per base sequence quality</p>
<p>横轴代表位置，纵轴quality，quality=-10*log10(p)，p为测错的概率。所以一条reads某位置出错概率为0.01时，其quality就是20。</p>
<p>红色横线表示中位数，蓝色折线是平均数；黄色块是25%-75%区间，上下的触须是10%-90%区间（即类似于箱式图)；</p>
<p>若任一位置的下四分位数低于10或中位数低于25，报”WARN”；若任一位置的下四分位数低于5或中位数低于20，报”FAIL”.</p>
<p><img src="https://i.loli.net/2020/12/25/bGTvZn18isoANFr.png" alt="per_base_quality"></p>
<p>3)每条reads的quality的均值的分布</p>
<pre><code>横轴为quality，纵轴是reads数目。当峰值小于27（错误率0.2%）时报"WARN"，当峰值小于20（错误率1%）时报"FAIL"。</code></pre>
<p><img src="https://i.loli.net/2020/12/25/vnigIpXtaVhxj6T.png" alt="per_sequence_quality"></p>
<p>②查看frags.B_fastqc.html</p>
<p>1)基本信息</p>
<p><img src="https://i.loli.net/2020/12/25/5DAIqCfmsYj2yMe.png"></p>
<p>2)Per base sequence quality</p>
<p><img src="https://i.loli.net/2020/12/25/Bl51ZvknPORDYTb.png"></p>
<p>3)3)每条reads的quality的均值的分布</p>
<p><img src="https://i.loli.net/2020/12/25/TCAEYd7roIa1pvB.png"></p>
<h3 id="2-5-使用AllPath-LG进行组装文库"><a href="#2-5-使用AllPath-LG进行组装文库" class="headerlink" title="2.5 使用AllPath-LG进行组装文库"></a>2.5 使用AllPath-LG进行组装文库</h3><h4 id="①将Fastq转换为ALLPATH-LG支持的输入格式"><a href="#①将Fastq转换为ALLPATH-LG支持的输入格式" class="headerlink" title="①将Fastq转换为ALLPATH-LG支持的输入格式"></a>①将Fastq转换为ALLPATH-LG支持的输入格式</h4><pre class=" language-bash"><code class="language-bash"><span class="token comment" spellcheck="true">#assembly</span>
<span class="token function">mkdir</span> -p <span class="token variable">$ALLPATH</span>/result.genome/data  <span class="token comment" spellcheck="true">#用于将转换后的数据文件放入到此目录下</span>

<span class="token comment" spellcheck="true">#inputs preparation</span>
<span class="token comment" spellcheck="true">#运行以下命令，将fastq文件转成运行ALLPATH-LG所需要的文件</span>
PrepareAllPathsInputs.pl \
        DATA_DIR<span class="token operator">=</span><span class="token variable">$ALLPATH</span>/result.genome/data \ <span class="token comment" spellcheck="true">#result.genome必须和csv文件中的organism_name相同</span>
        PLOIDY<span class="token operator">=</span>1 \ <span class="token comment" spellcheck="true">##生成ploidy文件，1表示基因组为单倍体型；2为双倍体型，allpaths-lg目前只支持单倍体和二倍体</span>
        IN_GROUPS_CSV<span class="token operator">=</span><span class="token variable">$DATA</span>/in_groups.csv \ <span class="token comment" spellcheck="true">#in_groups.csv用于指出测序数据的存放路径</span>
        IN_LIBS_CSV<span class="token operator">=</span><span class="token variable">$DATA</span>/in_libs.csv \ <span class="token comment" spellcheck="true"># in_libs.csv用于给出文库的特性</span>
        GENOME_SIZE<span class="token operator">=</span>200000\
        OVERWRITE<span class="token operator">=</span>True <span class="token operator">|</span><span class="token function">tee</span> <span class="token variable">$ALLPATH</span>/prepare.out <span class="token operator">>></span>/dev/null <span class="token comment" spellcheck="true">#是否覆盖存在的文件。可以设置该选项为True</span>
        <span class="token comment" spellcheck="true">#tee命令用于读取标准输入的数据，并将其内容输出成文件</span>
</code></pre>
<p> 在allpath/bin目录下，有一个名为PrepareAllPathsInputs.pl的可执行文件。</p>
<p><img src="https://i.loli.net/2020/12/25/zwCyokEfVKiZPpO.png"></p>
<p>这个文件就是用来准备输入文件的。这个文件需要读取以下两个文件：in_groups.csv和in_libs.csv</p>
<p>in_groups.csv文件的解释：</p>
<blockquote>
<p>group_name:数据独特的代号,每一份数据有一个代号；</p>
<p>library_name:数据所属文库的名字；</p>
<p>filename:数据文件所存放位置。可以为相对位置，文件名可以包含’*’和’?’(但是扩展名中不能有该符号，因为要根据扩展名识别文件类型)，从而代表paired数据。支持的文件类型有’.bam’,’fasta’,’fa’,’fastq’,’fq’,’fastq.gz’和’fq.gz’。</p>
</blockquote>
<p>in_libs.csv文件的解释：</p>
<blockquote>
<p>library_name:和in_groups.csv中的相匹配；</p>
<p>project_name:project的名字；</p>
<p>organism_name:测序物种的名字；</p>
<p>type:文库类型，fragment代表插入片段短，存在overlap的文库；jumping代表插入片段非常长的文库</p>
<p>paired: 0代表单端测序，1代表双端测序;</p>
<p>frag_size:小片段文库插入片段长度的均值；</p>
<p>frag_stddev:小片段文库的插入片段长度估算的标准偏差；</p>
<p>insert_size:大片段文库插入片段长度的均值；</p>
<p>insert_stddev:大片段文库插入片段长度估算的标准偏差；</p>
<p>read_orientation:reads的方向，小片段文库为inward，大片段文库为outward；</p>
<p>genomic_start:reads从该位置开始，读入数据，如果不为0，之前的碱基都被剪掉；</p>
<p>genomic_end:reads从该位置开始，停止读入数据，如果不为0，之后的碱基都被剪掉。</p>
</blockquote>
<p>运行结束后</p>
<p>使用<code>less $ALLPATH/prepare.out</code>查看输出信息</p>
<p><img src="https://i.loli.net/2020/12/25/qPR38kvQHbMs1hW.png"></p>
<p><img src="https://i.loli.net/2020/12/25/uv6lk2GD9Y8Z7eQ.png"></p>
<p>在输出目录会生成如下文件    </p>
<p>每个文库的序列会生成对应的.fastb, .pairs, qualb三个文件；ploidy 记录染色体倍性；read_cache 是临时目录。</p>
<p><img src="https://i.loli.net/2020/12/25/hDypLX6eCZEjm5S.png"></p>
<h4 id="②准备好输入文件后，使用RunAllPathsLG这个命令来进行基因组组装"><a href="#②准备好输入文件后，使用RunAllPathsLG这个命令来进行基因组组装" class="headerlink" title="②准备好输入文件后，使用RunAllPathsLG这个命令来进行基因组组装"></a>②准备好输入文件后，使用RunAllPathsLG这个命令来进行基因组组装</h4><pre class=" language-bash"><code class="language-bash">RunAllPathsLG \
        PRE<span class="token operator">=</span><span class="token variable">$ALLPATH</span> \ <span class="token comment" spellcheck="true">#程序运行的根目录，所有的其他目录全在该目录下</span>
        REFERENCE_NAME<span class="token operator">=</span>result.genome \ <span class="token comment" spellcheck="true">#参考基因组目录名称，位于PRE目录下，若有参考基因组，可放于此目录下</span>
        DATA_SUBDIR<span class="token operator">=</span>data \  <span class="token comment" spellcheck="true">#DATA子目录，位于REFERENCE_NAME目录下，程序从该目录下读取数据。</span>
        RUN<span class="token operator">=</span>run \ <span class="token comment" spellcheck="true">#位于DATA_SUBDIR目录下，将生成的中间文件和结果文件存储与该目录中。</span>
        SUBDIR<span class="token operator">=</span>test \
        TARGETS<span class="token operator">=</span>standard \
        OVERWRITE<span class="token operator">=</span>True <span class="token operator">|</span><span class="token function">tee</span> -a <span class="token variable">$ALLPATH</span>/assemble.out <span class="token operator">>></span>/dev/null
</code></pre>
<p>执行截图</p>
<p><img src="https://i.loli.net/2020/12/25/SWpDFcCznxf5MJj.png"></p>
<p>上述命令中的5个参数构成了如下的目录结构</p>
<p><code>PRE/REFERENCE_NAME/DATA_SUBDIR/RUN/SUBDIR</code></p>
<p>结果为<code>demo/genome_assembly/result/allpathlg/result.genome/data/run/ASSEMBLIES/test/final.*.fastg</code></p>
<h4 id="③查看基因组组装结果"><a href="#③查看基因组组装结果" class="headerlink" title="③查看基因组组装结果"></a>③查看基因组组装结果</h4><p>使用<code>ls final.*.fastg</code>查看有多少文件</p>
<p><img src="https://i.loli.net/2020/12/25/jMlkbfFJ1Qw75DB.png"></p>
<p>使用<code>less $ALLPATH/assemble.out</code>查看输出信息</p>
<p><img src="https://i.loli.net/2020/12/25/pat2ET4gkqJz5su.png"><img src="https://i.loli.net/2020/12/25/wHChB7xfWREDqSO.png"></p>
<p>使用<code>less final.contigs.fastg</code>查看final.contigs.fastg</p>
<p><img src="https://i.loli.net/2020/12/25/3vE9wh1DjL4QGlP.png"></p>
<p>使用<code>less final.assembly.fastg</code>查看final.assembly.fastg</p>
<p><img src="https://i.loli.net/2020/12/25/wjqo8FSMGNQ67lz.png"></p>
<h2 id="附：Tmux的使用"><a href="#附：Tmux的使用" class="headerlink" title="附：Tmux的使用"></a>附：Tmux的使用</h2><p>打开一个终端窗口，在里面输入命令，这种用户与计算机的临时交互，称为<strong>一次会话（session）</strong>。</p>
<p>由于我们使用的是ssh登录linux服务器，为了断电断网能够继续执行命令，则可以使用tmux来运行命令。</p>
<p>tmux就是会话与窗口的<strong>解绑工具</strong>，窗口关闭后会话并不终止，会继续运行，还允许在单个串口同时访问多个会话，方便多命令行的操作。</p>
<p><strong>tmux的主要命令</strong></p>
<pre class=" language-bash"><code class="language-bash"><span class="token comment" spellcheck="true"># 新建会话</span>
tmux new -s <span class="token operator">&lt;</span>session-name<span class="token operator">></span>
<span class="token comment" spellcheck="true">#命令可以查看当前所有的 Tmux 会话。</span>
tmux <span class="token function">ls</span>
<span class="token comment" spellcheck="true"># 重新接入某个已存在的会话</span>
tmux attach -t <span class="token operator">&lt;</span>session-name<span class="token operator">></span>
<span class="token comment" spellcheck="true">#杀死某个会话</span>
tmux kill-session -t <span class="token operator">&lt;</span>session-name<span class="token operator">></span></code></pre>
<p>主要快捷键：</p>
<ul>
<li><code>Ctrl+b d</code>：分离当前会话。</li>
<li><code>Ctrl+b s</code>：列出所有会话。</li>
<li><code>Ctrl+b $</code>：重命名当前会话。</li>
</ul>
<p>Tmux 的最简操作流程：</p>
<ol>
<li>新建会话<code>tmux new -s my_session</code>。</li>
<li>在 Tmux 窗口运行所需的程序。</li>
<li>按下快捷键<code>Ctrl+b d</code>将会话分离。</li>
<li>下次使用时，重新连接到会话<code>tmux attach-session -t my_session</code>。</li>
</ol>
<h2 id="参考资料"><a href="#参考资料" class="headerlink" title="参考资料"></a>参考资料</h2><ul>
<li><a target="_blank" rel="noopener" href="http://www.ruanyifeng.com/blog/2019/10/tmux.html">Tmux 使用教程 by 阮一峰</a></li>
<li><a target="_blank" rel="noopener" href="https://www.plob.org/article/9388.html">使用 GCE 进行基因组大小评估</a></li>
<li><a target="_blank" rel="noopener" href="https://www.plob.org/article/5987.html">用FastQC检查二代测序原始数据的质量</a></li>
<li><a target="_blank" rel="noopener" href="https://www.google.com/url?sa=t&amp;rct=j&amp;q=&amp;esrc=s&amp;source=web&amp;cd=&amp;ved=2ahUKEwiizN74v9nsAhWMxYsBHdXuDfwQFjACegQIBRAC&amp;url=http://www.chenlianfu.com/?tag=allpaths-lg&amp;usg=AOvVaw2tqfoBTFp--5c-LMqhWYpw">AllPaths-LG | 陈连福的生信博客</a></li>
<li><a target="_blank" rel="noopener" href="https://www.jianshu.com/p/d122522a1fd1">基因组De novo组装和 ALLPATHS-LG使用</a></li>
</ul>

            </div>
            <hr/>

            



            <div class="tag_share" style="display: block;">
                <div class="post-meta__tag-list" style="display: inline-block;">
                    
                        <div class="article-tag">
                            
                                <a href="/tags/%E7%94%9F%E4%BF%A1%E8%B5%84%E6%BA%90%E4%B8%8E%E5%AE%9E%E8%B7%B5/">
                                    <span class="chip bg-color">生信资源与实践</span>
                                </a>
                            
                        </div>
                    
                </div>
                <div class="post_share" style="zoom: 80%; width: fit-content; display: inline-block; float: right; margin: -0.15rem 0;">
                    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/share/css/share.min.css">
<div id="article-share">

    
    <div class="social-share" data-sites="twitter,facebook,google,qq,qzone,wechat,weibo,douban,linkedin" data-wechat-qrcode-helper="<p>微信扫一扫即可分享！</p>"></div>
    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/share/js/social-share.min.js"></script>
    

    

</div>

                </div>
            </div>
            
        </div>
    </div>

    

    

    

    

    
        <style>
    .valine-card {
        margin: 1.5rem auto;
    }

    .valine-card .card-content {
        padding: 20px 20px 5px 20px;
    }

    #vcomments textarea {
        box-sizing: border-box;
        background: url("/medias/comment_bg.png") 100% 100% no-repeat;
    }

    #vcomments p {
        margin: 2px 2px 10px;
        font-size: 1.05rem;
        line-height: 1.78rem;
    }

    #vcomments blockquote p {
        text-indent: 0.2rem;
    }

    #vcomments a {
        padding: 0 2px;
        color: #4cbf30;
        font-weight: 500;
        text-decoration: none;
    }

    #vcomments img {
        max-width: 100%;
        height: auto;
        cursor: pointer;
    }

    #vcomments ol li {
        list-style-type: decimal;
    }

    #vcomments ol,
    ul {
        display: block;
        padding-left: 2em;
        word-spacing: 0.05rem;
    }

    #vcomments ul li,
    ol li {
        display: list-item;
        line-height: 1.8rem;
        font-size: 1rem;
    }

    #vcomments ul li {
        list-style-type: disc;
    }

    #vcomments ul ul li {
        list-style-type: circle;
    }

    #vcomments table, th, td {
        padding: 12px 13px;
        border: 1px solid #dfe2e5;
    }

    #vcomments table, th, td {
        border: 0;
    }

    table tr:nth-child(2n), thead {
        background-color: #fafafa;
    }

    #vcomments table th {
        background-color: #f2f2f2;
        min-width: 80px;
    }

    #vcomments table td {
        min-width: 80px;
    }

    #vcomments h1 {
        font-size: 1.85rem;
        font-weight: bold;
        line-height: 2.2rem;
    }

    #vcomments h2 {
        font-size: 1.65rem;
        font-weight: bold;
        line-height: 1.9rem;
    }

    #vcomments h3 {
        font-size: 1.45rem;
        font-weight: bold;
        line-height: 1.7rem;
    }

    #vcomments h4 {
        font-size: 1.25rem;
        font-weight: bold;
        line-height: 1.5rem;
    }

    #vcomments h5 {
        font-size: 1.1rem;
        font-weight: bold;
        line-height: 1.4rem;
    }

    #vcomments h6 {
        font-size: 1rem;
        line-height: 1.3rem;
    }

    #vcomments p {
        font-size: 1rem;
        line-height: 1.5rem;
    }

    #vcomments hr {
        margin: 12px 0;
        border: 0;
        border-top: 1px solid #ccc;
    }

    #vcomments blockquote {
        margin: 15px 0;
        border-left: 5px solid #42b983;
        padding: 1rem 0.8rem 0.3rem 0.8rem;
        color: #666;
        background-color: rgba(66, 185, 131, .1);
    }

    #vcomments pre {
        font-family: monospace, monospace;
        padding: 1.2em;
        margin: .5em 0;
        background: #272822;
        overflow: auto;
        border-radius: 0.3em;
        tab-size: 4;
    }

    #vcomments code {
        font-family: monospace, monospace;
        padding: 1px 3px;
        font-size: 0.92rem;
        color: #e96900;
        background-color: #f8f8f8;
        border-radius: 2px;
    }

    #vcomments pre code {
        font-family: monospace, monospace;
        padding: 0;
        color: #e8eaf6;
        background-color: #272822;
    }

    #vcomments pre[class*="language-"] {
        padding: 1.2em;
        margin: .5em 0;
    }

    #vcomments code[class*="language-"],
    pre[class*="language-"] {
        color: #e8eaf6;
    }

    #vcomments [type="checkbox"]:not(:checked), [type="checkbox"]:checked {
        position: inherit;
        margin-left: -1.3rem;
        margin-right: 0.4rem;
        margin-top: -1px;
        vertical-align: middle;
        left: unset;
        visibility: visible;
    }

    #vcomments b,
    strong {
        font-weight: bold;
    }

    #vcomments dfn {
        font-style: italic;
    }

    #vcomments small {
        font-size: 85%;
    }

    #vcomments cite {
        font-style: normal;
    }

    #vcomments mark {
        background-color: #fcf8e3;
        padding: .2em;
    }

    #vcomments table, th, td {
        padding: 12px 13px;
        border: 1px solid #dfe2e5;
    }

    table tr:nth-child(2n), thead {
        background-color: #fafafa;
    }

    #vcomments table th {
        background-color: #f2f2f2;
        min-width: 80px;
    }

    #vcomments table td {
        min-width: 80px;
    }

    #vcomments [type="checkbox"]:not(:checked), [type="checkbox"]:checked {
        position: inherit;
        margin-left: -1.3rem;
        margin-right: 0.4rem;
        margin-top: -1px;
        vertical-align: middle;
        left: unset;
        visibility: visible;
    }
</style>

<div class="card valine-card" data-aos="fade-up">
    <div class="comment_headling" style="font-size: 20px; font-weight: 700; position: relative; padding-left: 20px; top: 15px; padding-bottom: 5px;">
        <i class="fas fa-comments fa-fw" aria-hidden="true"></i>
        <span>评论</span>
    </div>
    <div id="vcomments" class="card-content" style="display: grid">
    </div>
</div>

<script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/valine/av-min.js"></script>
<script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/valine/Valine.min.js"></script>
<script>
    new Valine({
        el: '#vcomments',
        appId: '6iI0iLCmPt2RF0cVJ4mpzk9g-MdYXbMMI',
        appKey: '65lnN4gsqHUBYBdA1ULByihM',
        notify: 'true' === 'true',
        verify: 'true' === 'true',
        visitor: 'true' === 'true',
        avatar: 'wavatar',
        pageSize: '10',
        lang: 'zh-cn',
        placeholder: '欢迎留下您的足迹'
    });
</script>

    

    

    

    

<article id="prenext-posts" class="prev-next articles">
    <div class="row article-row">
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge left-badge text-color">
                <i class="fas fa-chevron-left"></i>&nbsp;上一篇</div>
            <div class="card">
                <a href="/posts/649f.html">
                    <div class="card-image">
                        
                        
                        <img src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/medias/featureimages/8.jpg" class="responsive-img" alt="【生信资源与实践】第三次上机-miRNA-seq&amp;exome-seq">
                        
                        <span class="card-title">【生信资源与实践】第三次上机-miRNA-seq&amp;exome-seq</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            miRNA-seq是一种建库方法和数据处理上都稍有特殊的一种测序方法。外显子组测序（Exome sequencing）是指利用序列捕获技术将全基因组外显子区域DNA捕捉并富集后进行高通量测序的基因组分析方法。
                        
                    </div>
                    <div class="publish-info">
                        <span class="publish-date">
                            <i class="far fa-clock fa-fw icon-date"></i>2020-12-25
                        </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/" class="post-category">
                                    生物信息
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/%E7%94%9F%E4%BF%A1%E8%B5%84%E6%BA%90%E4%B8%8E%E5%AE%9E%E8%B7%B5/">
                        <span class="chip bg-color">生信资源与实践</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge right-badge text-color">
                下一篇&nbsp;<i class="fas fa-chevron-right"></i>
            </div>
            <div class="card">
                <a href="/posts/2c94.html">
                    <div class="card-image">
                        
                        
                        <img src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/medias/featureimages/3.jpg" class="responsive-img" alt="【生信资源与实践】第一次上机Linux基础命令&amp;RNA_seq">
                        
                        <span class="card-title">【生信资源与实践】第一次上机Linux基础命令&amp;RNA_seq</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            Linux 基础命令& vi & RNA_seq(Bowtie-tophat2-Cufflinks-Cuffdifff)
                        
                    </div>
                    <div class="publish-info">
                            <span class="publish-date">
                                <i class="far fa-clock fa-fw icon-date"></i>2020-12-25
                            </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF/" class="post-category">
                                    生物信息
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/%E7%94%9F%E4%BF%A1%E8%B5%84%E6%BA%90%E4%B8%8E%E5%AE%9E%E8%B7%B5/">
                        <span class="chip bg-color">生信资源与实践</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
    </div>
</article>

</div>



<!-- 代码块功能依赖 -->
<script type="text/javascript" src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/codeBlock/codeBlockFuction.js"></script>

<!-- 代码语言 -->

<script type="text/javascript" src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/codeBlock/codeLang.js"></script>


<!-- 代码块复制 -->

<script type="text/javascript" src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/codeBlock/codeCopy.js"></script>


<!-- 代码块收缩 -->

<script type="text/javascript" src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/codeBlock/codeShrink.js"></script>


<!-- 代码块折行 -->

<style type="text/css">
code[class*="language-"], pre[class*="language-"] { white-space: pre !important; }
</style>


    </div>
    <div id="toc-aside" class="expanded col l3 hide-on-med-and-down">
        <div class="toc-widget card" style="background-color: white;">
            <div class="toc-title"><i class="far fa-list-alt"></i>&nbsp;&nbsp;目录</div>
            <div id="toc-content"></div>
        </div>
    </div>
</div>

<!-- TOC 悬浮按钮. -->

<div id="floating-toc-btn" class="hide-on-med-and-down">
    <a class="btn-floating btn-large bg-color">
        <i class="fas fa-list-ul"></i>
    </a>
</div>


<script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/tocbot/tocbot.min.js"></script>
<script>
    $(function () {
        tocbot.init({
            tocSelector: '#toc-content',
            contentSelector: '#articleContent',
            headingsOffset: -($(window).height() * 0.4 - 45),
            collapseDepth: Number('0'),
            headingSelector: 'h2, h3, h4'
        });

        // modify the toc link href to support Chinese.
        let i = 0;
        let tocHeading = 'toc-heading-';
        $('#toc-content a').each(function () {
            $(this).attr('href', '#' + tocHeading + (++i));
        });

        // modify the heading title id to support Chinese.
        i = 0;
        $('#articleContent').children('h2, h3, h4').each(function () {
            $(this).attr('id', tocHeading + (++i));
        });

        // Set scroll toc fixed.
        let tocHeight = parseInt($(window).height() * 0.4 - 64);
        let $tocWidget = $('.toc-widget');
        $(window).scroll(function () {
            let scroll = $(window).scrollTop();
            /* add post toc fixed. */
            if (scroll > tocHeight) {
                $tocWidget.addClass('toc-fixed');
            } else {
                $tocWidget.removeClass('toc-fixed');
            }
        });

        
        /* 修复文章卡片 div 的宽度. */
        let fixPostCardWidth = function (srcId, targetId) {
            let srcDiv = $('#' + srcId);
            if (srcDiv.length === 0) {
                return;
            }

            let w = srcDiv.width();
            if (w >= 450) {
                w = w + 21;
            } else if (w >= 350 && w < 450) {
                w = w + 18;
            } else if (w >= 300 && w < 350) {
                w = w + 16;
            } else {
                w = w + 14;
            }
            $('#' + targetId).width(w);
        };

        // 切换TOC目录展开收缩的相关操作.
        const expandedClass = 'expanded';
        let $tocAside = $('#toc-aside');
        let $mainContent = $('#main-content');
        $('#floating-toc-btn .btn-floating').click(function () {
            if ($tocAside.hasClass(expandedClass)) {
                $tocAside.removeClass(expandedClass).hide();
                $mainContent.removeClass('l9');
            } else {
                $tocAside.addClass(expandedClass).show();
                $mainContent.addClass('l9');
            }
            fixPostCardWidth('artDetail', 'prenext-posts');
        });
        
    });
</script>

    

</main>




    <footer class="page-footer bg-color">
    
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/aplayer/APlayer.min.css">
<style>
    .aplayer .aplayer-lrc p {
        
        display: none;
        
        font-size: 12px;
        font-weight: 700;
        line-height: 16px !important;
    }

    .aplayer .aplayer-lrc p.aplayer-lrc-current {
        
        display: none;
        
        font-size: 15px;
        color: #42b983;
    }

    
    .aplayer.aplayer-fixed.aplayer-narrow .aplayer-body {
        left: -66px !important;
    }

    .aplayer.aplayer-fixed.aplayer-narrow .aplayer-body:hover {
        left: 0px !important;
    }

    
</style>
<div class="">
    
    <div class="row">
        <meting-js class="col l8 offset-l2 m10 offset-m1 s12"
                   server="netease"
                   type="playlist"
                   id="136126210"
                   fixed='true'
                   autoplay='false'
                   theme='#42b983'
                   loop='all'
                   order='random'
                   preload='auto'
                   volume='0.5'
                   list-folded='true'
        >
        </meting-js>
    </div>
</div>

<script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/aplayer/APlayer.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/meting@2/dist/Meting.min.js"></script>

    
    <div class="container row center-align"
        style="margin-bottom: 15px !important;">
        <div class="col s12 m8 l8 copy-right">
            Copyright&nbsp;&copy;
            
            <span id="year">2020</span>
            
            <span id="year">2020</span>
            <a href="/about" target="_blank">风渡巛</a>
            |&nbsp;Powered by&nbsp;<a href="https://hexo.io/" target="_blank">Hexo</a>
            |&nbsp;Theme&nbsp;<a href="https://github.com/blinkfox/hexo-theme-matery" target="_blank">Matery</a>
            <br>
            
            &nbsp;<i class="fas fa-chart-area"></i>&nbsp;站点总字数:&nbsp;<span
                class="white-color">45.8k</span>&nbsp;字
            
            
            
            
            
            
            <span id="busuanzi_container_site_pv">
                |&nbsp;<i class="far fa-eye"></i>&nbsp;总访问量:&nbsp;<span id="busuanzi_value_site_pv"
                    class="white-color"></span>&nbsp;次
            </span>
            
            
            <span id="busuanzi_container_site_uv">
                |&nbsp;<i class="fas fa-users"></i>&nbsp;总访问人数:&nbsp;<span id="busuanzi_value_site_uv"
                    class="white-color"></span>&nbsp;人
            </span>
            
            <br>
            
            <span id="sitetime">载入运行时间...</span>
            <script>
                function siteTime() {
                    var seconds = 1000;
                    var minutes = seconds * 60;
                    var hours = minutes * 60;
                    var days = hours * 24;
                    var years = days * 365;
                    var today = new Date();
                    var startYear = "2020";
                    var startMonth = "11";
                    var startDate = "14";
                    var startHour = "0";
                    var startMinute = "0";
                    var startSecond = "0";
                    var todayYear = today.getFullYear();
                    var todayMonth = today.getMonth() + 1;
                    var todayDate = today.getDate();
                    var todayHour = today.getHours();
                    var todayMinute = today.getMinutes();
                    var todaySecond = today.getSeconds();
                    var t1 = Date.UTC(startYear, startMonth, startDate, startHour, startMinute, startSecond);
                    var t2 = Date.UTC(todayYear, todayMonth, todayDate, todayHour, todayMinute, todaySecond);
                    var diff = t2 - t1;
                    var diffYears = Math.floor(diff / years);
                    var diffDays = Math.floor((diff / days) - diffYears * 365);
                    var diffHours = Math.floor((diff - (diffYears * 365 + diffDays) * days) / hours);
                    var diffMinutes = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours) /
                        minutes);
                    var diffSeconds = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours -
                        diffMinutes * minutes) / seconds);
                    if (startYear == todayYear) {
                        document.getElementById("year").innerHTML = todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已艰难运行 " + diffDays + " 天 " + diffHours +
                            " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    } else {
                        document.getElementById("year").innerHTML = startYear + " - " + todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已艰难运行 " + diffYears + " 年 " + diffDays +
                            " 天 " + diffHours + " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    }
                }
                setInterval(siteTime, 1000);
            </script>
            
            <br>
            
        </div>
        <div class="col s12 m4 l4 social-link social-statis">
    <a href="https://github.com/Achuan-2" class="tooltipped" target="_blank" data-tooltip="访问我的GitHub" data-position="top" data-delay="50">
        <i class="fab fa-github"></i>
    </a>



    <a href="mailto:270992395@qq.com" class="tooltipped" target="_blank" data-tooltip="邮件联系我" data-position="top" data-delay="50">
        <i class="fas fa-envelope-open"></i>
    </a>







    <a href="tencent://AddContact/?fromId=50&fromSubId=1&subcmd=all&uin=270992395" class="tooltipped" target="_blank" data-tooltip="QQ联系我: 270992395" data-position="top" data-delay="50">
        <i class="fab fa-qq"></i>
    </a>





    <a href="https://www.zhihu.com/people/su-ji-xiong-62" class="tooltipped" target="_blank" data-tooltip="关注我的知乎: https://www.zhihu.com/people/su-ji-xiong-62" data-position="top" data-delay="50">
        <i class="fab fa-zhihu1">知</i>
    </a>


    <a href="https://space.bilibili.com/349243695" class="tooltipped" target="_blank" data-tooltip="关注我的B站: https://space.bilibili.com/349243695" data-position="top" data-delay="50">
        <i >B</i>
    </a>




    <a href="/atom.xml" class="tooltipped" target="_blank" data-tooltip="RSS 订阅" data-position="top" data-delay="50">
        <i class="fas fa-rss"></i>
    </a>

</div>
    </div>
</footer>

<div class="progress-bar"></div>

    <!-- 搜索遮罩框 -->
<div id="searchModal" class="modal">
    <div class="modal-content">
        <div class="search-header">
            <span class="title"><i class="fas fa-search"></i>&nbsp;&nbsp;搜索</span>
            <input type="search" id="searchInput" name="s" placeholder="请输入搜索的关键字"
                   class="search-input">
        </div>
        <div id="searchResult"></div>
    </div>
</div>

<script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/js/search.js"></script>
<script type="text/javascript">
$(function () {
    searchFunc('/search.xml', 'searchInput', 'searchResult');
});
</script>

    <!-- 回到顶部按钮 -->
<div id="backTop" class="top-scroll">
    <a class="btn-floating btn-large waves-effect waves-light" href="#!">
        <i class="fas fa-arrow-up"></i>
    </a>
</div>


    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/materialize/materialize.min.js"></script>
    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/masonry/masonry.pkgd.min.js"></script>
    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/aos/aos.js"></script>
    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/scrollprogress/scrollProgress.min.js"></script>
    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/lightGallery/js/lightgallery-all.min.js"></script>
    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/js/matery.js"></script>

    <!-- Baidu Analytics -->

    <!-- Baidu Push -->

<script>
    (function () {
        var bp = document.createElement('script');
        var curProtocol = window.location.protocol.split(':')[0];
        if (curProtocol === 'https') {
            bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
        } else {
            bp.src = 'http://push.zhanzhang.baidu.com/push.js';
        }
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(bp, s);
    })();
</script>

    
    
    <script async src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/others/busuanzi.pure.mini.js"></script>
    

    

    

    
    

    

    

    
    <script src="https://cdn.jsdelivr.net/gh/Achuan-2/Achuan-2.github.io/libs/instantpage/instantpage.js" type="module"></script>
    
    <!--动态线条背景-->
    <script type="text/javascript" color="122 103 238" opacity='0.7' zIndex="-2" count="200"
        src="//cdn.bootcss.com/canvas-nest.js/1.0.0/canvas-nest.min.js">
        </script>

<script src="/live2dw/lib/L2Dwidget.min.js?094cbace49a39548bed64abff5988b05"></script><script>L2Dwidget.init({"model":{"scale":1,"hHeadPos":0.5,"vHeadPos":0.618,"jsonPath":"/live2dw/assets/koharu.model.json"},"display":{"superSample":2,"position":"right","width":150,"height":300,"hOffset":-30,"vOffset":60},"mobile":{"show":false,"scale":0.5},"react":{"opacity":0.7,"opacityOnHover":0.2},"log":false,"pluginJsPath":"lib/","pluginModelPath":"assets/","pluginRootPath":"live2dw/","tagMode":false});</script></body>

</html>